src/pkcs11/framework-pkcs15.c

```diff
@@ -3410,6 +3410,22 @@ pkcs15_gen_keypair(struct sc_pkcs11_slot *slot, CK_MECHANISM_PTR pMechanism,
 		keygen_args.prkey_args.user_consent = 1;
 	}
 
+	/* set EXTRACTABLE and SENSITIVESE flags */
+	rv = attr_find(pPrivTpl, ulPrivCnt, CKA_EXTRACTABLE, &always_auth, &len);
+	if (rv == CKR_OK && always_auth == CK_TRUE)
+		keygen_args.prkey_args.access_flags |= SC_PKCS15_PRKEY_ACCESS_EXTRACTABLE;
+	else
+		keygen_args.prkey_args.access_flags |= SC_PKCS15_PRKEY_ACCESS_NEVEREXTRACTABLE;
+
+	rv = attr_find(pPrivTpl, ulPrivCnt, CKA_SENSITIVE, &always_auth, &len);
+	if (rv == CKR_OK && always_auth == CK_TRUE) {
```

Comment:
please, use a different variable than `always_auth` here. It makes the code very confusing. Either separate variable for both of these or just `bool_value` for all of them together.

Reply:
I will make separate variables here.